### PR TITLE
Made Apple HTTP client using async requests.

### PR DESCRIPTION
**Pull request by mzhurovich**
@dkorolev I love Apple. PTAL.

### CODE REVIEW

File: Bricks/net/api/impl/apple_client.mm [hunk 1]
```diff
@@ -86,30 +89,39 @@ of this software and associated documentation files (the "Software"), to deal
...
-      if (received_file.empty()) {
-        server_response.assign(reinterpret_cast<char const *>(url_data.bytes), url_data.length);
-      } else {
-        if (![url_data writeToFile:[NSString stringWithUTF8String:received_file.c_str()] atomically:YES]) {
-          return false;
-        }
-      }
-    }
-    return true;
+    __block bool request_succeeded = true;
```
Comment (dkorolev) [inline Bricks/net/api/impl/apple_client.mm]: Why not move this magic into the request object too?

File: Bricks/net/api/impl/apple_client.mm [hunk 2]
```diff
@@ -86,30 +89,39 @@ of this software and associated documentation files (the "Software"), to deal
...
+              if (![data writeToFile:[NSString stringWithUTF8String:received_file.c_str()] atomically:YES]) {
+                request_succeeded = false;
+              }
+            }
+          }
+          *async_request_completed.MutableScopedAccessor() = true;
+    }];
+
+    async_request_completed.Wait([](bool done) { return done; });
+    return request_succeeded;
```
Comment (dkorolev) [inline Bricks/net/api/impl/apple_client.mm]: Blank vertical whitespace.

**Comment by mzhurovich** [comment]
@dkorolev Thnx! Take another look, pls.

**Comment by dkorolev** [comment]
LGTM++!

**Pull request merged**